core/runtime/binaryen/module/wasm_module_impl.cpp

```diff
@@ -58,14 +63,31 @@ namespace kagome::runtime::binaryen {
       } catch (wasm::ParseException &e) {
         std::ostringstream msg;
         e.dump(msg);
-        auto log = log::createLogger("wasm_module", "wasm");
         log->error(msg.str());
         return Error::INVALID_STATE_CODE;
       }
     }
 
-    module->memory.initial = 16_MB / 64_kB;  // 64kB pages for 16Mb
-
+    module->memory.initial = kDefaultHeappages;
+    OUTCOME_TRY(heappages_key, common::Buffer::fromString(":heappages"));
+    if (auto heappages_res =
+            storage_provider->getCurrentBatch()->get(heappages_key);
```

Comment:
Did not see the else-branch for the condition. Is it ok?

Reply:
Yes, the default value is set above the "if".

Reply:
Perhaps it is better to pass only "Not found" error, not others ones